components/scream/src/physics/spa/tests/spa_main_test.cpp

```diff
@@ -352,57 +350,25 @@ TEST_CASE("spa_read_data","spa")
-    const auto& ccn3_in = ekat::subview(ccn3_beg,dof_i);
-    compute_max_min(ccn3_in,1,spa_data_beg.nlevs-1,ccn3_bnds_h(0,dof_i),ccn3_bnds_h(1,dof_i));
-    for (int n=0;n<nswbands;n++) {
-      const auto& sw_g_in = ekat::subview(aer_g_sw_beg,dof_i,n);
-      compute_max_min(sw_g_in,1,spa_data_beg.nlevs-1,aer_sw_g_bnds_h(0,dof_i,n),aer_sw_g_bnds_h(1,dof_i,n));
-      const auto& sw_ssa_in = ekat::subview(aer_ssa_sw_beg,dof_i,n);
-      compute_max_min(sw_ssa_in,1,spa_data_beg.nlevs-1,aer_sw_ssa_bnds_h(0,dof_i,n),aer_sw_ssa_bnds_h(1,dof_i,n));
-      const auto& sw_tau_in = ekat::subview(aer_tau_sw_beg,dof_i,n);
-      compute_max_min(sw_tau_in,1,spa_data_beg.nlevs-1,aer_sw_tau_bnds_h(0,dof_i,n),aer_sw_tau_bnds_h(1,dof_i,n));
-    }
-    for (int n=0;n<nlwbands;n++) {
-      const auto& lw_tau_in = ekat::subview(aer_tau_lw_beg,dof_i,n);
-      compute_max_min(lw_tau_in,1,spa_data_beg.nlevs-1,aer_lw_tau_bnds_h(0,dof_i,n),aer_lw_tau_bnds_h(1,dof_i,n));
-    }
-  }
+  using col_type = view_1d<Real>::HostMirror;
+  auto check_bounds = [&](const col_type& data,const col_type& output) {
+    auto data_minmax   = compute_max_min(data,0,nlevs+1);
+    auto output_minmax = compute_max_min(output,0,nlevs);
+    REQUIRE( data_minmax.first  >= output_minmax.first );
+    REQUIRE( data_minmax.second >= output_minmax.second );
+  };
 
-  // Make sure the output data is within the same bounds as the input data.
   for (size_t dof_i=0;dof_i<dofs_gids_h.size();dof_i++) {
-    for (int kk=0;kk<nlevs;kk++) {
-      int kpack = kk / Spack::n;
-      int kidx  = kk % Spack::n;
-      REQUIRE(ccn3_h(dof_i,kpack)[kidx]>=ccn3_bnds_h(0,dof_i));
-      REQUIRE(ccn3_h(dof_i,kpack)[kidx]<=ccn3_bnds_h(1,dof_i));
-      for (int n=0;n<nswbands;n++) {
-        REQUIRE(aer_g_sw_h(dof_i,n,kpack)[kidx]   >= aer_sw_g_bnds_h  (0,dof_i,n) );
-        REQUIRE(aer_ssa_sw_h(dof_i,n,kpack)[kidx] >= aer_sw_ssa_bnds_h(0,dof_i,n) );
-        REQUIRE(aer_tau_sw_h(dof_i,n,kpack)[kidx] >= aer_sw_tau_bnds_h(0,dof_i,n) );
-        REQUIRE(aer_g_sw_h(dof_i,n,kpack)[kidx]   <= aer_sw_g_bnds_h  (1,dof_i,n) );
-        REQUIRE(aer_ssa_sw_h(dof_i,n,kpack)[kidx] <= aer_sw_ssa_bnds_h(1,dof_i,n) );
-        REQUIRE(aer_tau_sw_h(dof_i,n,kpack)[kidx] <= aer_sw_tau_bnds_h(1,dof_i,n) );
-      }
-      for (int n=0;n<nlwbands;n++) {
-        REQUIRE(aer_tau_lw_h(dof_i,n,kpack)[kidx] >= aer_lw_tau_bnds_h(0,dof_i,n) );
-        REQUIRE(aer_tau_lw_h(dof_i,n,kpack)[kidx] <= aer_lw_tau_bnds_h(1,dof_i,n) );
-      }
+    const auto& ccn3_in  = ekat::subview(ekat::scalarize(ccn3_beg),dof_i);
+    const auto& ccn3_out = ekat::subview(ekat::scalarize(ccn3_h),dof_i);
+    check_bounds( ccn3_in, ccn3_out );
+    for (int n=0;n<nswbands;n++) {
+      auto ssv = [&](const view_3d<Spack>& v, const int dof_i, const int n) -> col_type {
+        return ekat::subview(ekat::scalarize(v),dof_i,n);
+      };
+      check_bounds (ssv(aer_ssa_sw_h,dof_i,n),ssv(aer_ssa_sw_beg,dof_i,n));
```

Comment:
Aren't you missing a similar line for all other band quantities?

Reply:
duh

Reply:
done